File: snake.js
// Initialize canvas and context
const canvas = document.getElementById('canvas');
const ctx = canvas.getContext('2d');

// Set canvas dimensions
const canvasWidth = 600;
const canvasHeight = 400;
canvas.width = canvasWidth;
canvas.height = canvasHeight;

// Set game variables
const blockSize = 10;
let snake = [{x: 10, y: 10}];
let food = {x: 0, y: 0};
let score = 0;
let direction = 'right';

// Generate random food location
function generateFood() {
  food.x = Math.floor(Math.random() * (canvasWidth / blockSize)) * blockSize;
  food.y = Math.floor(Math.random() * (canvasHeight / blockSize)) * blockSize;
}

// Draw the snake and food on the canvas
function draw() {
  // Clear the canvas
  ctx.clearRect(0, 0, canvasWidth, canvasHeight);

  // Draw the snake
  ctx.fillStyle = '#4CAF50';
  snake.forEach(segment => {
    ctx.fillRect(segment.x, segment.y, blockSize, blockSize);
  });

  // Draw the food
  ctx.fillStyle = '#FF5722';
  ctx.fillRect(food.x, food.y, blockSize, blockSize);
}

// Move the snake
function move() {
  // Move the snake's body
  for (let i = snake.length - 1; i > 0; i--) {
    snake[i].x = snake[i - 1].x;
    snake[i].y = snake[i - 1].y;
  }

  // Move the snake's head
  switch (direction) {
    case 'up':
      snake[0].y -= blockSize;
      break;
    case 'down':
      snake[0].y += blockSize;
      break;
    case 'left':
      snake[0].x -= blockSize;
      break;
    case 'right':
      snake[0].x += blockSize;
      break;
  }
}

// Check for collisions with the walls and the snake's own body
function checkCollisions() {
  // Wrap the snake's position around to the opposite side of the canvas if it goes out of bounds
  if (snake[0].x < 0) {
    snake[0].x = canvasWidth - blockSize;
  } else if (snake[0].x >= canvasWidth) {
    snake[0].x = 0;
  } else if (snake[0].y < 0) {
    snake[0].y = canvasHeight - blockSize;
  } else if (snake[0].y >= canvasHeight) {
    snake[0].y = 0;
  }

  // Check for collision with the snake's own body
  for (let i = 1; i < snake.length; i++) {
    if (snake[0].x === snake[i].x && snake[0].y === snake[i].y) {
      gameOver();
    }
  }
}

// Check if the snake has eaten the food
function checkFood() {
  if (snake[0].x === food.x && snake[0].y === food.y) {
    // Increase the score
    score++;

    // Add a new segment to the snake
    snake.push({x: snake[snake.length - 1].x, y: snake[snake.length - 1].y});

    // Generate new food
    generateFood();
  }
}

// Game over function
function gameOver() {
  alert(`Game over! Your score is ${score}.`);

  // Reset game variables
  snake = [{x: 10, y: 10}];
  food = {x: 0, y: 0};
  score = 0;
  direction = 'right';

  // Generate new food
  generateFood();
}

// Handle keyboard input
document.addEventListener('keydown', event => {
  switch (event.keyCode) {
    case 37: // Left arrow
      if (direction !== 'right') {
        direction = 'left';
      }
      break;
    case 38: // Up arrow
      if (direction !== 'down') {
        direction = 'up';
      }
      break;
    case 39: // Right arrow
      if (direction !== 'left') {
        direction = 'right';
      }
      break;
    case 40: // Down arrow
      if (direction !== 'up') {
        direction = 'down';
      }
      break;
  }
});

// Game loop
function gameLoop() {
  move();
  checkCollisions();
  checkFood();
  draw();
}

// Start the game loop
generateFood();
setInterval(gameLoop, 100);

File: snake.py
import random
import pygame


# 初始化Pygame
pygame.init()

# 游戏窗口大小
WINDOW_WIDTH = 800
WINDOW_HEIGHT = 600

# 颜色定义
BLACK = (0, 0, 0)
WHITE = (255, 255, 255)
RED = (255, 0, 0)
GREEN = (0, 255, 0)
BLUE = (0, 0, 255)

# 创建游戏窗口
window = pygame.display.set_mode((WINDOW_WIDTH, WINDOW_HEIGHT))
pygame.display.set_caption('Snake Game')

# 定义贪吃蛇类
class Snake:
    def __init__(self):
        self.body = [[WINDOW_WIDTH // 2, WINDOW_HEIGHT // 2]]
        self.direction = 'right'

    def move(self):
        x, y = self.body[0]
        if self.direction == 'up':
            y -= 10
        elif self.direction == 'down':
            y += 10
        elif self.direction == 'left':
            x -= 10
        elif self.direction == 'right':
            x += 10
        self.body.insert(0, [x, y])
        self.body.pop()

    def draw(self):
        for x, y in self.body:
            pygame.draw.rect(window, GREEN, (x, y, 10, 10))

    def grow(self):
        x, y = self.body[0]
        if self.direction == 'up':
            y -= 10
        elif self.direction == 'down':
            y += 10
        elif self.direction == 'left':
            x -= 10
        elif self.direction == 'right':
            x += 10
        self.body.insert(0, [x, y])

    def check_collision(self):
        x, y = self.body[0]
        if x < 0 or x > WINDOW_WIDTH or y < 0 or y > WINDOW_HEIGHT:
            return True
        for i in range(1, len(self.body)):
            if self.body[i] == self.body[0]:
                return True
        return False

# 定义食物类
class Food:
    def __init__(self):
        self.x = random.randint(0, WINDOW_WIDTH - 10)
        self.y = random.randint(0, WINDOW_HEIGHT - 10)

    def draw(self):
        pygame.draw.rect(window, RED, (self.x, self.y, 10, 10))

# 创建贪吃蛇和食物对象
snake = Snake()
food = Food()

# 游戏循环
running = True
while running:
    # 处理事件
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.KEYDOWN:
            if event.key == pygame.K_UP and snake.direction != 'down':
                snake.direction = 'up'
            elif event.key == pygame.K_DOWN and snake.direction != 'up':
                snake.direction = 'down'
            elif event.key == pygame.K_LEFT and snake.direction != 'right':
                snake.direction = 'left'
            elif event.key == pygame.K_RIGHT and snake.direction != 'left':
                snake.direction = 'right'

    # 移动贪吃蛇
    snake.move()

    # 检查是否吃到食物
    if snake.body[0] == [food.x, food.y]:
        snake.grow()
        food = Food()

    # 检查是否碰到边界或自身
    if snake.check_collision():
        running = False

    # 绘制游戏界面
    window.fill(BLACK)
    snake.draw()
    food.draw()
    pygame.display.update()

# 退出Pygame
pygame.quit()
